velox/exec/RowNumber.cpp

```diff
@@ -489,4 +508,46 @@ void RowNumber::spillInput(
   }
 }
 
+BlockingReason RowNumber::isBlocked(ContinueFuture* /* unused */) {
+  const auto reason =
+      yield_ ? BlockingReason::kYield : BlockingReason::kNotBlocked;
```

Comment:
This logic is a bit strange. isBlocked is supposed to set the 'future' if operator is blocked, but this code doesn't do that.

Reply:
We need to return BlockingReason::kYield in `isBlocked` to make the driver framework aware to off the thread and yield the CPU.
https://github.com/facebookincubator/velox/blob/main/velox/exec/Driver.cpp#L545
https://github.com/facebookincubator/velox/blob/main/velox/exec/Driver.cpp#L776

Reply:
Fixed. Yes, it is kind of hacky. Turns out we could leverage the **yield** in the driver framework itself.
https://github.com/facebookincubator/velox/blob/main/velox/exec/Driver.cpp#L526

Reply:
@duanmeng Thanks. Should we add a check that if isBlocked returns something other than kNotBlocked, then future is set? (in a separate PR)

Reply:
@mbasmanova I am not sure whether we need this check as there is no blocking future in `RowNumber` set the 'future' of the `isBlocked` API.

Reply:
I was thinking that the caller of the Operator::isBlocked API may need to add a check that if the API returned "blocked", then the future is set.

Reply:
Do you mean we need to add a check in the Driver::runInternal to ensure that if the `op->isBlocked` does not return `kNotBlocked` then the `future` is not empty? So that this can regulate developers to ensure that the future is correctly set when overriding the `isBlocked` API.

Reply:
This is what I'm thinking about.